### PR TITLE
hal: pinctrl

### PR DESCRIPTION
**Pull request by glaubermaroto**

### CODE REVIEW

**Review by uLipe — APPROVED** [review]
Generally LGTM

File: dts/espressif/32/pinctrl.yaml [hunk 1]
```diff
@@ -0,0 +1,101 @@
+# According to ESP32's TRM:
```
Comment (sylvioalves) [inline dts/espressif/32/pinctrl.yaml:1]: Does it need copyright?

File: dts/espressif/32/pinctrl.yaml [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+# GPIO pads 34-39 are input­only.
+
+uart0:
+  tx:
+    func: [out]
+    gpio: [[0, 19], [21, 23], [25, 27], [32, 33]]
+    sigo: u0txd_out
+  rx:
+    func: [in]
+    bias: pull_up
```
Comment (sylvioalves) [inline dts/espressif/32/pinctrl.yaml:13]: Please, check which pins for PU/PD pins. AFAIK, ESP32 pins 34~39 have no PU/PD pins. This should be checked for all SoCs.

File: zephyr/scripts/esp_genpinctrl.py [hunk 3]
```diff
@@ -0,0 +1,200 @@
+#!/usr/bin/python3
```
Comment (sylvioalves) [inline zephyr/scripts/esp_genpinctrl.py:1]: Does it need copyright?

**Pull request merged**